cg/server/endpoints/orders.py

```diff
@@ -272,31 +264,11 @@ def validate_order(order_type: OrderType):
     raw_order = request.get_json()
     raw_order["project_type"] = order_type
     raw_order["user_id"] = g.current_user.id
-    response = {}
-    if order_type == OrderType.BALSAMIC:
-        response = balsamic_validation_service.validate(raw_order)
-    elif order_type == OrderType.BALSAMIC_UMI:
-        response = balsamic_umi_validation_service.validate(raw_order)
-    elif order_type == OrderType.FASTQ:
-        response = fastq_validation_service.validate(raw_order)
-    elif order_type == OrderType.METAGENOME:
-        response = metagenome_validation_service.validate(raw_order)
-    elif order_type == OrderType.MICROBIAL_FASTQ:
-        response = microbial_fastq_validation_service.validate(raw_order)
-    elif order_type == OrderType.MICROSALT:
-        response = microsalt_validation_service.validate(raw_order)
-    elif order_type == OrderType.MIP_DNA:
-        response = mip_dna_validation_service.validate(raw_order)
-    elif order_type == OrderType.PACBIO_LONG_READ:
-        response = pacbio_long_read_validation_service.validate(raw_order)
-    elif order_type == OrderType.SARS_COV_2:
-        response = mutant_validation_service.validate(raw_order)
-    elif order_type == OrderType.RML:
-        response = rml_validation_service.validate(raw_order)
-    elif order_type == OrderType.RNAFUSION:
-        response = rna_fusion_validation_service.validate(raw_order)
-    elif order_type == OrderType.TAXPROFILER:
-        response = taxprofiler_validation_service.validate(raw_order)
-    elif order_type == OrderType.TOMTE:
-        response = tomte_validation_service.validate(raw_order)
+    rule_set: RuleSet = ORDER_TYPE_RULE_SET_MAP[order_type]
+    model: type[Order] = ORDER_TYPE_MODEL_MAP[order_type]
+    response = order_validation_service.get_validation_response(
+        raw_order=raw_order,
+        rule_set=rule_set,
+        model=model,
+    )
```

Comment:
❤️

Reply:
💓

Reply:
as discussed on the huddle- the rule set and model can be parsed within the service. The service signature can receive just the order_type.

this way the endpoint does not require to know the information of a rule set or model map

Reply:
Changed!